CalibTracker/SiPixelESProducers/plugins/alpaka/SiPixelCablingSoAESProducer.cc

```diff
@@ -0,0 +1,161 @@
+#include "CondFormats/DataRecord/interface/SiPixelFedCablingMapRcd.h"
+#include "CondFormats/DataRecord/interface/SiPixelQualityRcd.h"
+#include "CondFormats/SiPixelObjects/interface/SiPixelFedCablingMap.h"
+#include "CondFormats/SiPixelObjects/interface/SiPixelFedCablingTree.h"
+#include "CondFormats/SiPixelObjects/interface/SiPixelQuality.h"
+#include "DataFormats/SiPixelClusterSoA/interface/gpuClusteringConstants.h"
+#include "CalibTracker/SiPixelESProducers/interface/SiPixelMappingHost.h"
+#include "CalibTracker/SiPixelESProducers/interface/alpaka/SiPixelMappingDevice.h"
+#include "FWCore/Framework/interface/ESTransientHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/ESGetToken.h"
+
+#include "Geometry/CommonDetUnit/interface/GeomDetType.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/ESProducer.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/ModuleFactory.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/memory.h"
+
+#include "RecoTracker/Record/interface/CkfComponentsRecord.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/CopyToDevice.h"
+#include "DataFormats/Portable/interface/alpaka/PortableCollection.h"
+
+template struct cms::alpakatools::CopyToDevice<SiPixelMappingHost>;  //needed for the method to not be incomplete
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  using namespace cms::alpakatools;
+
+  class SiPixelCablingSoAESProducer : public ESProducer {
+  public:
+    SiPixelCablingSoAESProducer(edm::ParameterSet const& iConfig)
+        : ESProducer(iConfig), useQuality_(iConfig.getParameter<bool>("UseQualityInfo")) {
+      auto const& component = iConfig.getParameter<std::string>("ComponentName");
+      auto cc = setWhatProduced(this, component);
```

Comment:
Not exactly (the detector description is somewhat special case where chains of ESProducers need consume and produce using the same product label; note that in the example you pointed to the `appendToDataLabel` from the configuration is used in `ESInputTag` that is consumed).

In the ESProducer side you can do just
```cpp
auto cc = setWhatProduced(this);
```
and in the configuration you get `appendToDataLabel` parameter automatically, i.e. you can do
```py
siPixelCablingSoAESProducer.appendToDataLabel = cms.string("blah")
# instead of
siPixelCablingSoAESProducer.ComponentName = cms.string("blah")
```
(note that if you specify both, the label would be "blahblah")